File: src/main/app/src/NGroup.js
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {HexUtils} from 'react-hexgrid';

class NGroup extends Component {

    static propTypes = {
        group: PropTypes.object,
        cellStyle: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.object
        ])
    };

    static contextTypes = {
        layout: PropTypes.object
    };

    render() {
        const {group, cellStyle} = this.props;
        const {layout} = this.context;
        const fill = group.color;
        const fillId = (fill) ? `#${fill}` : null;
        const start = group.start;
        const end = group.end;
        const size = 40;
        let startPoint = HexUtils.hexToPixel(start, layout);
        let endPoint = HexUtils.hexToPixel(end, layout);
        let width = endPoint.x - startPoint.x + 4 * size;
        let height = endPoint.y - startPoint.y + 5 * size;
        return (
            <g>
                <rect x={startPoint.x - 2 * size} y={startPoint.y - 2 * size} rx="50" ry="50" width={width}
                      height={height}
                      style={cellStyle}
                      stroke={fillId} fill={fillId} className="group"/>
                <text x={startPoint.x - 2 * size + width / 2} y={startPoint.y + height - 3 * size} fill={fillId}
                      font-size="24" width={width} textAnchor={'middle'}>{group.name}</text>
            </g>
        );
    }
}

export default NGroup;

File: src/main/java/de/bonndan/nivio/output/map/NGroup.java
package de.bonndan.nivio.output.map;

import j2html.tags.DomContent;

class NGroup extends Component {

    private final GroupMapItem group;
    private final int size;
    private final double x;
    private final double y;
    private final int width;
    private final int height;

    NGroup(GroupMapItem group, double x, double y, int width, int height) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
        size = 40;
        this.group = group;
    }

    public DomContent render() {
        var fill = group.color;
        var fillId = (fill) != null ? "#" + fill : null;

        return SvgTagCreator.g(
                SvgTagCreator.rect()
                        .attr("x", x)
                        .attr("y", y)
                        .attr("rx", 50)
                        .attr("ry", 50)
                        .attr("width", width)
                        .attr("height", height)
                        //.attr("style", cellStyle)
                        .attr("stroke", fillId)
                        .attr("fill", fillId)
                        .attr("class", "group"),

                SvgTagCreator.text(group.name)
                        .attr("x", x+ size/2)
                        .attr("y", y + height + size/2)
                        .attr("fill", fillId)
                        .attr("font-size", 24)
                        .attr("width", width)
                        .attr("text-anchor", "left")
                        .attr("class", "groupLabel")
        );
    }
}

